cmd/tcpgoon.go

```diff
@@ -1,15 +1,90 @@
 package cmd
 
 import (
+	"errors"
 	"fmt"
+	"os"
+	"strconv"
 
 	"github.com/dachad/tcpgoon/cmdutil"
 	"github.com/dachad/tcpgoon/debugging"
 	"github.com/dachad/tcpgoon/mtcpclient"
 	"github.com/dachad/tcpgoon/tcpclient"
+	"github.com/spf13/cobra"
 )
 
-func run(params tcpgoonParams) {
+type tcpgoonParams struct {
+	hostPtr              string
+	portPtr              int
+	numberConnectionsPtr int
+	delayPtr             int
+	connDialTimeoutPtr   int
+	debugPtr             bool
+	reportingIntervalPtr int
+	assumeyesPtr         bool
+}
+
+var params tcpgoonParams
+
+var rootCmd = &cobra.Command{
+	Use:   "tcpgoon <host> <port>",
+	Short: "tcpgoon tests concurrent connections towards a server listening on a TCP port",
+	Long:  ``,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		if err := validateRequiredArgs(&params, args); err != nil {
+			cmd.Println(cmd.UsageString())
+			os.Exit(1)
+		}
+		enableDebuggingIfFlagSet(params)
+		autorunValidation(params)
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		runroot(params)
+	},
+}
+
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	rootCmd.Flags().IntVarP(&params.numberConnectionsPtr, "connections", "c", 100, "Number of connections you want to open")
+	rootCmd.Flags().IntVarP(&params.delayPtr, "sleep", "s", 10, "Time you want to sleep between connections, in ms")
+	rootCmd.Flags().IntVarP(&params.connDialTimeoutPtr, "dial-timeout", "d", 5000, "Connection dialing timeout, in ms")
+	rootCmd.Flags().BoolVarP(&params.debugPtr, "verbose", "v", false, "Print debugging information to the standard error")
+	rootCmd.Flags().IntVarP(&params.reportingIntervalPtr, "interval", "i", 1, "Interval, in seconds, between stats updates")
+	rootCmd.Flags().BoolVarP(&params.assumeyesPtr, "assume-yes", "y", false, "Force execution without asking for confirmation")
+}
+
+func validateRequiredArgs(params *tcpgoonParams, args []string) error {
+	if len(args) != 2 {
+		return errors.New("Number of required parameters doesn't match")
+	}
+	params.hostPtr = args[0]
+	if port, err := strconv.Atoi(args[1]); err != nil && port <= 0 {
+		return errors.New("Port argument is not a valid integer")
+	} else {
```

Comment:
Improvement captured by codacy: "if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)" -> this means you don't need the else block. Move params.portPtr = port after the if, as if the condition matches it means it will exit before reaching that point

Reply:
ok, I've needed to move the strconv out of the if to make this possible, if not, a warning because port not assigned was raising